### PR TITLE
edits 404 page and (I hope) makes it work

### LINKED ISSUE #126: 404 page
404 Not Found errors fail silently:
You get the content of the home page,
but the URL *appears to work* ---
it doesn't redirect,
nothing tells you that the page you are getting is:

 - the front page
 - not the page you were looking for

This is a bit confusing.

### PR DESCRIPTION
**Pull request by adammichaelwood**
Closes #126 

#### What is included in this PR?

There was already a 404 page, but it never appeared
(Even locally) 
unless you navigated to it on purpose.

I edited it, adding a list of helpful links.
It is very sparse, but I think it gets the job done.

Then I updated `s3_website.yml` to show that page as the `error_document`.
I didn't find this documented anywhere,
but Googling around showed this is what a lot of projects' `s3_website` file looks like. 

#### What problems did you encounter?

 - My edit to `s3_website` is an educated guess.
 - The local server that comes with `jekyll serve` doesn't handle the 404 either.

So, basically, I don't know if this is right.

#### What is left to be done in the addressed issue?

Make sure it works.

### CODE REVIEW

**Comment by yanokwa** [comment]
Agreed with @wbrunette. Made the changes and I'll merge to see if it works.

**Comment by adammichaelwood** [comment]
I like the inline form,
but I would have left the Docs and Forum links there also.

Someone coming in on a broken link is *probably* looking for something that's in the docs.

**Comment by yanokwa** [comment]
@adammichaelwood My rationale is that the person doesn't have the context required to know what ODK vs ODK 2, and that using the links up top is their best bet.

**Comment by yanokwa** [comment]
Just FYI for a future person who has this problem. I had to fix 404 by manually entering it in the S3 UI.  My guess is that s3_website's cfg wasn't being applied for some reason.
